src/Assignments/ProblemIframe.tsx

```diff
@@ -348,7 +355,7 @@ export const ProblemIframe: React.FC<ProblemIframeProps> = ({
             // check that the submit url is accurate
             const submitUrl = problemForm.getAttribute('action');
             const checkId = submitUrl?.match(/\/backend-api\/courses\/question\/([0-9]+)\?/);
-            if (checkId && parseInt(checkId[1],10) !== problem.id) {
+            if (!checkId || parseInt(checkId[1],10) !== problem.id) {
```

Comment:
@doombot-exe this was the proposed change in slack

Reply:
@doombot-exe said he is against this in cases like static problem where there is no url period
I still think it should be here, however I'm fine with doing:
```
((submitUrl && !checkId) || parseInt(checkId[1],10) !== problem.id))
```

Reply:
it's not that the submitUrl won't exist though... I'm only saying that it's not worth checking the problemId if the submitUrl doesn't match the RegEx. We _will_ have issues where the iframe renders content that doesn't belong to a "problem" -- I'm thinking of problem editing or elsewhere that has previews without being "formally" a topic-problem.